src/nu/validator/saxtree/Node.java

```diff
@@ -75,13 +81,15 @@
         if (locator == null) {
             this.systemId = null;
             this.publicId = null;
+            this.encoding = null;
             this.column = -1;
             this.line = -1;
         } else {
             this.systemId = locator.getSystemId();
             this.publicId = locator.getPublicId();
             this.column = locator.getColumnNumber();
             this.line = locator.getLineNumber();
+            this.encoding = ((Locator2)locator).getEncoding();
```

Comment:
Please put this behind an `instanceof` check to keep saxtree generally useful.

Reply:
Done